prov/usnic/src/usdf_eq.c

```diff
@@ -194,11 +197,28 @@ static ssize_t usdf_eq_readerr(struct fid_eq *feq,
 		goto done;
 	}
 
-	ret = usdf_eq_read_event(eq, NULL, entry, sizeof(*entry), flags);
+	ret = usdf_eq_read_event(eq, NULL, &entry, sizeof(entry), flags);
+
+	/* read the user's setting for err_data. */
+	err_data = buffer->err_data;
+	err_data_size = buffer->err_data_size;
+
+	/* Copy the entry. */
+	*buffer = entry;
+
+	/* For compatibility purposes, if err_data_size is 0 on input, or the fab-
+	 * ric was opened with release < 1.5, err_data will be set to a data
+	 * buffer owned by the provider. */
+	api_version = eq->eq_fabric->fab_attr.api_version;
+	if (FI_VERSION_GE(api_version, FI_VERSION(1,5))) {
+		buffer->err_data = err_data;
+		buffer->err_data_size = MIN(err_data_size, entry.err_data_size);
```

Comment:
I think you should probably add a debug message here if the data has been truncated